src/main/java/v/hryhoryk/onlinebookstore/service/shoppingcart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package v.hryhoryk.onlinebookstore.service.shoppingcart;
+
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import v.hryhoryk.onlinebookstore.dto.cartitemdto.CartItemDto;
+import v.hryhoryk.onlinebookstore.dto.cartitemdto.CartItemQuantityDto;
+import v.hryhoryk.onlinebookstore.dto.cartitemdto.CartItemRequestDto;
+import v.hryhoryk.onlinebookstore.dto.shoppingcartdto.ShoppingCartDto;
+import v.hryhoryk.onlinebookstore.exceptions.EntityNotFoundException;
+import v.hryhoryk.onlinebookstore.mapper.CartItemMapper;
+import v.hryhoryk.onlinebookstore.mapper.ShoppingCartMapper;
+import v.hryhoryk.onlinebookstore.model.CartItem;
+import v.hryhoryk.onlinebookstore.model.ShoppingCart;
+import v.hryhoryk.onlinebookstore.model.User;
+import v.hryhoryk.onlinebookstore.repository.book.BookRepository;
+import v.hryhoryk.onlinebookstore.repository.cartitem.CartItemRepository;
+import v.hryhoryk.onlinebookstore.repository.shoppingcart.ShoppingCartRepository;
+import v.hryhoryk.onlinebookstore.security.CustomUserDetailsService;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CustomUserDetailsService userDetailsService;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public ShoppingCartDto getShoppingCartByUserId(Authentication authentication) {
+        User user = (User) userDetailsService.loadUserByUsername(authentication.getName());
```

Comment:
I would suggest to create private method for getting User to avoid duplicating code